File: src/m/dev/devpath.m
% clear the last warning to focus on the warnings of the ISSM path
lastwarn(''); 

%Recover ISSM_DIR , or if on a Windows machine, ISSM_DIR_WIN
if ispc,
	ISSM_DIR=getenv('ISSM_DIR_WIN');
else
	ISSM_DIR=getenv('ISSM_DIR');
end

if (isempty(ISSM_DIR)),
	error('''ISSM_DIR'' environment variable is empty! You should define ISSM_DIR in your .cshrc or .bashrc!');
end

%Now add all issm code paths necessary to run issm smoothly. 
%We capture the error output, so that we can warn the user to update 
%the variable ISSM_DIR in this file, in case it is not correctly setup. 

%ISSM path
addpath([ISSM_DIR '/src/m/os/']);       %load recursivepath
addpath([ISSM_DIR '/lib']);             %load MEX files
addpath(recursivepath([ISSM_DIR '/src/m']));
addpath(recursivepath([ISSM_DIR '/externalpackages/scotch']));
addpath(recursivepath([ISSM_DIR '/externalpackages/canos']));
addpath(recursivepath([ISSM_DIR '/externalpackages/kml']));
addpath(recursivepath([ISSM_DIR '/externalpackages/export_fig']));
addpath(recursivepath([ISSM_DIR '/externalpackages/googleearthtoolbox']));
addpath(recursivepath([ISSM_DIR '/externalpackages/howatmask']));
addpath(recursivepath([ISSM_DIR '/externalpackages/dem']));
addpath(recursivepath([ISSM_DIR '/externalpackages/mealpix']));
addpath(recursivepath([ISSM_DIR '/externalpackages/pcatool']));

%Check on any warning messages that might indicate that the paths were not correct. 
if ~isempty(lastwarn),
	fprintf('\n  Error trying to setup ''ISSM'' code paths. Try and update the ISSM_DIR variable in your .cshrc or .bashrc!\n');
	fprintf('  ''ISSM'' will not  work at all until this is resolved\n\n');
else
	fprintf('\n  ISSM development path correctly loaded\n\n');
end

warning ('off','all');
addpath([ISSM_DIR '/lib-precompiled']); %load MEX files (precompiled; remove after MEX file compilation is supported on Silicon-based Macs)
warning ('on','all');
clear ISSM_DIR;

%disable matlab bell!
beep off;

File: src/m/dev/devpath.py
#!/usr/bin/env python3
import os
import sys

# Recover ISSM_DIR and USERNAME
ISSM_DIR = os.getenv('ISSM_DIR')
USERNAME = os.getenv('USER')
if ISSM_DIR is None:
    raise NameError('"ISSM_DIR" environment variable is empty. You should define ISSM_DIR in your .zshrc or .bashrc')

# Go through src/m and append any directory that contains a *.py file to PATH
for root, dirs, files in os.walk(ISSM_DIR + '/src/m'):
    if '.svn' in dirs:
        dirs.remove('.svn')
    for file in files:
        if file.find('.py') != -1:
            if file.find('.pyc') == -1:
                if root not in sys.path:
                    sys.path.append(root)

# Also add the Nightly run directory
if ISSM_DIR + '/test/NightlyRun' not in sys.path:
    sys.path.append(ISSM_DIR + '/test/NightlyRun')
if ISSM_DIR + '/lib' not in sys.path:
    sys.path.append(ISSM_DIR + '/lib')
if ISSM_DIR + '/src/wrappers/python/.libs' not in sys.path:
    sys.path.append(ISSM_DIR + '/src/wrappers/python/.libs')

try: # Avoid circular import
    from runme import runme  # First, because plotmodel may fail
except:
    pass
from plotmodel import plotmodel

#c = get_ipython().config
#c.InteractiveShellApp.exec_lines = []
#c.InteractiveShellApp.exec_lines.append('%load_ext autoreload')
#c.InteractiveShellApp.exec_lines.append('%autoreload 2')
#c.InteractiveShellApp.exec_lines.append('print "Warning: disable autoreload in startup.py to improve performance." ')

# print("\n  ISSM development path correctly loaded")
# print("Current path is {}\n\n".format(ISSM_DIR))

